sphinx/themes/basic/static/searchtools.js

```diff
@@ -351,12 +351,14 @@ const Search = {
 
     // remove duplicate search results
     // note the reversing of results, so that in the case of duplicates, the highest-scoring entry is kept
+    // this may remove some entries which refer to slightly different parts of the same document
+    // but this is a tradeoff to avoid showing the same document multiple times (the preview always looks the
+    // same, so this is not very useful)
     let seen = new Set();
     results = results.reverse().reduce((acc, result) => {
-      let resultStr = result.slice(0, 4).concat([result[5]]).map(v => String(v)).join(',');
```

Comment:
Another naive question: could this have worked as an alternative?

```javascript
let resultStr = result[0] || '') + (result[2] || '');
```

It feels to me like `docname` plus `anchor` should be a unique key.  But I could be mistaken.

Reply:
I suspect so, based on my reading of the code and debugging. It feels like these keys have a great deal of redundant information in them. Unfortunately there's no pull request associated with the original commit adding this: 

https://github.com/sphinx-doc/sphinx/commit/e3a8744235adeefdecc1160c0fadfa2322c4bf7d

Reply:
Digging in deeper, it appears there's also a "descr" field that was used. I'm not fully confident it's safe to take this out.

Reply:
Update: I left `descr` in as a sorting key. It appears the only case where a search result has one is the case of an "object" search:

https://github.com/wlach/sphinx/blob/949e13142c75bd4801ee6f6f1767a32a0a5e1338/sphinx/themes/basic/static/searchtools.js#L446

I suspect we probably don't need this to be in the search key for de-duplication purposes, but seems safest to just leave this part of the logic unchanged.

Reply:
Yes, better to reduce the possible breaking changes